lms/static/scripts/frontend_apps/components/test/LMSGrader-test.js

```diff
@@ -0,0 +1,78 @@
+import { act } from 'preact/test-utils';
+import { Fragment, createElement } from 'preact';
+import { mount } from 'enzyme';
+import LMSGrader, { $imports } from '../LMSGrader';
+
+describe('LMSGrader', () => {
+  const fakeStudents = [
+    {
+      userid: 'user1',
+      displayName: 'User 1',
+    },
+    {
+      userid: 'user2',
+      displayName: 'User 2',
+    },
+  ];
+  const fakeUpdateClientConfig = sinon.spy();
+  const fakeOnChange = sinon.stub();
+
+  // eslint-disable-next-line react/prop-types
+  const FakeStudentSelector = ({ children }) => {
+    return <Fragment>{children}</Fragment>;
+  };
+
+  beforeEach(() => {
+    $imports.$mock({
+      '../utils/update-client-config': fakeUpdateClientConfig,
+      './StudentSelector': FakeStudentSelector,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  const renderGrader = (props = {}) => {
+    return mount(
+      <LMSGrader
+        onChangeSelectedUser={fakeOnChange}
+        students={fakeStudents}
+        {...props}
+      />
+    );
+  };
+
+  it('does not set a focus user by default', () => {
+    renderGrader();
+    sinon.assert.calledWith(
+      fakeUpdateClientConfig,
+      sinon.match({
+        focus: {},
+      })
+    );
+  });
+
+  it('changes the sidebar config to focus to the specified user when onSelectStudent is called with a valid user index', () => {
```

Comment:
What happens if `onSelectStudent` is called with an invalid user index?

Reply:
That bit of logic is handled in the `useEffect` on the state var change, and handled in the first test in this file. I can call out a unique test for this, but it won't test any more lines of actau code but may make logical sense when reading the tests.